ale_linters/eruby/erb.vim

```diff
@@ -0,0 +1,24 @@
+" Author: Matthias Guenther - https://wikimatze.de, Eddie Lebow https://github.com/elebow
+" Description: ERB from the Ruby standard library, for eruby/erb files
+
+function! ale_linters#eruby#erb#GetCommand(buffer) abort
+    let l:rails_root = ale#ruby#FindRailsRoot(a:buffer)
+
+    if l:rails_root is? ''
+        return 'erb -P -x %t | ruby -c'
+    endif
+
+    " Rails-flavored eRuby does not comply with the standard as understood by
+    " ERB, so we'll have to do some substitution. This does not reduce the
+    " effectiveness of the linter—the translated code is still evaluated.
+    return 'ruby -rerb -e "puts ERB.new(File.read(%t, encoding: \"BINARY\").gsub(\"<%=\",\"<%\"), nil, \"-\").src" | ruby -c'
```

Comment:
The escaping of double quotes here won't work on Windows. I recommend removing the `\"` escapes, and use `ale#Escape()` for the parameter instead. That should work everywhere. Same for the file below.

Reply:
`ale#Escape('puts ERB.new(File.read(%t, encoding: "BINARY").gsub("<%=","<%"), nil, "-").src')` That might be what you want to add to the string there.

Reply:
Oh, and the percents will also be doubled on Windows. Windows is weird.

Reply:
I just realised, the `%t` inside of the escaped string might cause issues where the escaping doesn't work any more when the filename contains spaces. If you can pass in the filename with an argument and use `ARGV[0]` or something, then do that.

Reply:
You might also be able to do something like `'rbuy -rerb -e filename=%t' . ale#Escape(';puts ERB.new(File.read(filename, ...')`

Reply:
I'm having trouble getting `ale#Escape()` to do what is needed here... it inserts quotes liberally, which is fine for filenames in a shell, but not in the middle of Ruby code.
But I was able to rewrite the line to not have any escaped double quotes:

```
 return 'ruby -r erb -e "puts ERB.new(\$stdin.read.gsub(%{<%=},%{<%}), nil, %{-}).src" | ruby -c'
```
And it works correctly on GNU/Linux. Is this okay for Windows? Do the `%` signs need to be escaped?

Reply:
Try this.

```vim
return 'ruby -r erb -e ' . ale#Escape('puts ERB.new($stdin.read.gsub(%{<%=},%{<%}), nil, %{-}).src') . ' | ruby -c'
```

The percent signs will probably need to be doubled on Windows.